L/LLVM@8/build_tarballs.jl

```diff
@@ -0,0 +1,251 @@
+    export ac_cv_have_decl___builtin_ffs=yes
+
+    # We don't use X-ray on BSD systems
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_XRAY=OFF)
+fi
+
+if [[ "${target}" == *mingw* ]]; then
+    CMAKE_CPP_FLAGS="${CMAKE_CPP_FLAGS} -remap -D__USING_SJLJ_EXCEPTIONS__ -D__CRT__NO_INLINE"
+    # Windows is case-insensitive and some dependencies take full advantage of that
+    echo "BaseTsd.h basetsd.h" >> /opt/${target}/${target}/include/header.gcc
+fi
+
+if [[ "${target}" == *musl* ]]; then
+    # Taken from https://git.alpinelinux.org/cgit/aports/tree/main/compiler-rt/APKBUILD
+    CMAKE_FLAGS+=(-DCOMPILER_RT_INCLUDE_TESTS=ON)
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_SANITIZERS=OFF)
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_XRAY=OFF)
+fi
+
+if [[ "${target}" == *freebsd* ]]; then
+    # On FreeBSD, we must force even statically-linked code to have -fPIC
+    CMAKE_FLAGS+=(-DCMAKE_POSITION_INDEPENDENT_CODE=TRUE)
+fi
+
+# Build!
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
+cmake -LA || true
+ninja -j${nproc}
+
+# Install!
+ninja install -j${nproc}
+
+# move clang products out of $prefix/bin to $prefix/tools
+mv ${prefix}/bin/clang* ${prefix}/tools/
+mv ${prefix}/bin/scan-* ${prefix}/tools/
+mv ${prefix}/bin/c-index* ${prefix}/tools/
+mv ${prefix}/bin/git-clang* ${prefix}/tools/
+mv ${prefix}/bin/lld* ${prefix}/tools/
+
+# Life is harsh on Windows and dynamic libraries are
+# expected to live alongside the binaries. So we have
+# to copy the *.dll from bin/ to tools/ as well...
+if [[ "${target}" == *mingw* ]]; then
+    cp ${prefix}/bin/*.dll ${prefix}/tools/
+fi
+
+# Work around llvm-config bug by creating versioned symlink to libLLVM
+# https://github.com/JuliaLang/julia/pull/30033
+if [[ "${target}" == *darwin* ]]; then
+    LLVM_VER=$(basename $(echo ${prefix}/tools/clang-*.*))
+    ln -s libLLVM.dylib ${prefix}/lib/libLLVM-${LLVM_VER##*-}.dylib
+fi
+
+# Lit is a python dependency and there is no proper install target
+cp -r ${LLVM_SRCDIR}/utils/lit ${prefix}/tools/
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:i686,    libc = :glibc),
+    Linux(:x86_64,  libc = :glibc),
+    Linux(:x86_64,  libc = :musl),
+    Linux(:aarch64, libc = :glibc),
+    Linux(:armv7l,  libc = :glibc),
+    Linux(:powerpc64le, libc = :glibc),
+    MacOS(),
+    Windows(:i686),
+    Windows(:x86_64),
+    FreeBSD(:x86_64),
+]
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+# We want libLLVM.so as well but it is called LLVM.dll on Windows
+products = [
+  LibraryProduct("libclang", :libclang)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+]
+
+
+build_tarballs(ARGS, "LLVM", version, sources, script, platforms, products, dependencies,
+               preferred_gcc_version=v"7", preferred_clang_version=v"7")
```

Comment:
Isn't it called `preferred_llvm_version`?

Reply:
You are right, very odd that it fixed a bug for me then ...

Reply:
Maybe it was some other change made at the same time?  I can't grep any `preferred_clang_version` in BinaryBuilder code.